control/ts2021/client.go

```diff
@@ -162,131 +144,71 @@ func NewNoiseClient(opts NoiseOpts) (*NoiseClient, error) {
 		logf:         opts.Logf,
 		netMon:       opts.NetMon,
 		health:       opts.HealthTracker,
+		dialSem:      make(chan struct{}, 1),
+		rawRT: &http.Transport{
+			IdleConnTimeout: 90 * time.Second,
+			Protocols:       new(http.Protocols),
+		},
 	}
 
-	// Create the HTTP/2 Transport using a net/http.Transport
-	// (which only does HTTP/1) because it's the only way to
-	// configure certain properties on the http2.Transport. But we
-	// never actually use the net/http.Transport for any HTTP/1
-	// requests.
-	h2Transport, err := http2.ConfigureTransports(&http.Transport{
-		IdleConnTimeout: time.Minute,
-	})
-	if err != nil {
-		return nil, err
+	// We force only HTTP/2 for this transport, which is what the control server
+	// speaks inside the ts2021 Noise encryption. But Go doesn't know about that,
+	// so we use "SetUnencryptedHTTP2" even though it's actually encrypted.
+	np.rawRT.Protocols.SetUnencryptedHTTP2(true)
+	np.rawRT.DialTLSContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		select {
+		case np.dialSem <- struct{}{}:
+			defer func() { <-np.dialSem }()
+		default:
+			// Another dial is in progress. Fail this dial and the
+			// caller RoundTrip, which will make NoiseClient.RoundTrip
+			// look and retry, hopefully reusing the existing connection
+			// once it's known to be HTTP/2.
+			return nil, errConcurrentNoiseDial
+		}
+		return np.dial(ctx)
```

Comment:
TODO: I want to add tests around this first.

I really wish the Go standard library had MaxConcurrentDialsPerHost

It does have MaxConnsPerHost at least. Maybe that's enough and we just set it at 1?

We should never need more than 2 connections total anyway. Hmmm.

Reply:
Done! Much simpler.